src/relayer/publicDecrypt.ts

```diff
@@ -42,6 +44,104 @@ function isThresholdReached(
   return recoveredAddresses.length >= threshold;
 }
 
+function abiEncodeClearValues(clearValues: ClearValues) {
+  const handlesBytes32Hex = Object.keys(clearValues) as `0x${string}`[];
+
+  const abiTypes: string[] = [];
+  const abiValues: (string | bigint)[] = [];
+
+  for (let i = 0; i < handlesBytes32Hex.length; ++i) {
+    const handle = handlesBytes32Hex[i];
+    const handleType = getHandleType(handle);
+
+    let clearTextValue: ClearValueType =
+      clearValues[handle as keyof typeof clearValues];
+    if (typeof clearTextValue === 'boolean') {
+      clearTextValue = clearTextValue ? '0x01' : '0x00';
+    }
+
+    const clearTextValueBigInt = BigInt(clearTextValue);
+
+    //abiTypes.push(fhevmTypeInfo.solidityTypeName);
+    abiTypes.push('uint256');
+
+    switch (handleType) {
+      // eaddress
+      case 7: {
+        // string
+        abiValues.push(
+          `0x${clearTextValueBigInt.toString(16).padStart(40, '0')}`,
+        );
+        break;
+      }
+      // ebool
+      case 0: {
+        // bigint (0 or 1)
+        if (
+          clearTextValueBigInt !== BigInt(0) &&
```

Comment:
The bounds are validated for `bool` type and not others. Could you share if there is a specific reason why we treat `bool` specially ?